lib/binding_web/src/query.ts

```diff
@@ -198,19 +211,322 @@ export class Query {
   /** The maximum number of in-progress matches for this cursor. */
   matchLimit?: number;
 
-  /** @internal */
-  constructor(
-    internal: Internal,
-    address: number,
-    captureNames: string[],
-    captureQuantifiers: CaptureQuantifier[][],
-    textPredicates: TextPredicate[][],
-    predicates: QueryPredicate[][],
-    setProperties: QueryProperties[],
-    assertedProperties: QueryProperties[],
-    refutedProperties: QueryProperties[],
-  ) {
-    assertInternal(internal);
+  /**
+   * Create a new query from a string containing one or more S-expression
+   * patterns.
+   *
+   * The query is associated with a particular language, and can only be run
+   * on syntax nodes parsed with that language. References to Queries can be
+   * shared between multiple threads.
+   *
+   * @link {@see https://tree-sitter.github.io/tree-sitter/using-parsers/queries}
+   */
+  constructor(language: Language, source: string) {
```

Comment:
Moving all this to `query.ts` is a good change, and I hope we can iteratively improve on this code in the future by _incrementally_ breaking all the work in this constructor into smaller, more digestible bits. It doesn't have to happen in this PR.

Reply:
You also don't _have_ to deprecate `Language.query`.

Reply:
> You also don't have to deprecate Language.query.

Why? It doesn't make sense to have two different ways to do the same thing, coming from two different classes. No other bindings do this